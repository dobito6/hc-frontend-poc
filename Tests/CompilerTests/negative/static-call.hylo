fun f0(x: Void) {}

fun f1<T>(x: T) { }

fun f2<T :: * -> *>() -> T {}

fun main() {
  f0<Void>(x: ())

  f1<Void>(x: ())
  f1<Void>(x: ((),))
  f1<Void, Void>(x: ())
}
