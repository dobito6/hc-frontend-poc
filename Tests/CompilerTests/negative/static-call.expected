negative/static-call.hylo:5.26-27: error: T expects 1 type argument(s)
fun f2<T :: * -> *>() -> T {}
                         ^
negative/static-call.hylo:8.3-5: error: 'f0' takes no compile-time arguments
  f0<Void>(x: ())
  ~~
negative/static-call.hylo:11.15-20: error: cannot pass value of type '{Void}' to parameter 'Void'
  f1<Void>(x: ((),))
              ~~~~~
negative/static-call.hylo:12.3-5: error: 'f1' takes 1 compile-time argument(s); found 2
  f1<Void, Void>(x: ())
  ~~
